lib/pages/list.js

```diff
@@ -0,0 +1,207 @@
+"use strict";
+var Promise = require("bluebird");
+var React = require("react");
+var Link = require("react-router").Link;
+var apiMapper = require("../logic/api-mapper");
+
+// Keys in state.pulls
+const unassignedKey = "unassigned";
+const awaitingReviewKey = "awaitingReview";
+const awaitingActionKey = "awaitingAction";
+const reviewedKey = "reviewed";
+const lgtmedKey = "lgtmed";
+
+// Keys in state.assignedPulls
+const assignedNoActionKey = "assignedNoAction";
+const assignedNeedsActionKey = "assignedNeedsAction";
+const assignedOthersInvolvedKey = "assignedOthersInvolvedAction";
+
+// States
+const unloaded = "unloaded";
+const loaded = "loaded";
+
+// Values of _renderPRs's display param
+const displayOwner = "displayOwner";
+
+module.exports = React.createClass({displayName: 'ListPage',
+    propTypes: {
+        controller: React.PropTypes.any,
+        sessionStore: React.PropTypes.any
+    },
+
+    getInitialState: function() {
+        return {
+            pulls: {},
+            assignedPulls: {},
+            pullsState: unloaded,
+            assignedPullsState: unloaded,
+            error: null
+        };
+    },
+
+    loadPulls: function() {
+        var self = this;
+        self.props.controller.httpApi.getUser()
+            .done(function(userData) {
+                self.props.controller.httpApi.getOpenPullRequests(userData.body.login, "author")
+                .then(self._fillComments)
+                .then(self._loadMyPulls)
+                .catch(function(err) {
+                    console.error(JSON.stringify(err));
+                    self.setState({
+                        error: "Unexpected error loading pull requests"
+                    });
+                });
+
+                self.props.controller.httpApi.getOpenPullRequests(userData.body.login, "assignee")
+                .then(self._fillComments)
+                .then(self._loadAssignedPulls)
+                .catch(function(err) {
+                    console.error(JSON.stringify(err));
+                    self.setState({
+                        error: "Unexpected error loading pull requests"
+                    });
+                });
+            }, function(err) {
+            if (err.statusCode == 401) {
+                self.setState({error: "Error getting user details - try entering a valid access token"});
+            } else {
+                self.setState({error: "Unexpected error getting user details"});
+            }
+        });
+    },
+
+    _loadMyPulls: function(prs) {
+        var pulls = {};
+        pulls[unassignedKey] = [];
+        pulls[awaitingReviewKey] = [];
+        pulls[awaitingActionKey] = [];
+        pulls[reviewedKey] = [];
+        pulls[lgtmedKey] = [];
```

Comment:
Oh oh I think you can use ES6 uhhh computed property keys? Maybe. Possibly. To steal from [MDN's excellent docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer):

```javascript
var o = {
  [prop]: "hey",
  ["b" + "ar"]: "there",
};
```